addons/mail/static/src/core/common/thread_model.js

```diff
@@ -1072,6 +1080,7 @@ export class Thread extends Record {
         if (pushState === undefined) {
             pushState = this.notEq(this.store.discuss.thread);
         }
+        this.markAsRead();
```

Comment:
Same remark as with `_openChatWindow`.
Also I'm not entirely sure `setAsDiscussThread` necessarily means the user is actively in the discuss app. So another reason to not manage here but instead the showing of thread message list is the actual trigger of `markAsRead`

Reply:
The message list would mean the thread component. I don't think we should couple the `localSeen` message state to the thread component life cycle. IMO it should be done on explicit user actions: the user opens a chat window or a discuss thread. An example of problematic scenario could be:
- Open a chat window, mark a message as unread
- Click on "Open in discuss" in the chat window header menu
- Separator is shown as expected
- Go back to the chat window
- Separator should still be shown. If we were to bind this state to the message list itself, the separator would not remain (the message list was indeed shown twice)

Reply:
I'm not saying this should simply be coupled to lifecycle of component. You even have a counter-argument with how components are defined: `ChatWindow <=> Discuss` transition are implemented as different component entities so this is a problem. However, I would still solve this issue by keeping a single point of `markAsRead()` rather than pretend that this action necessarily means the message will be marked as read.

Nothing could have prevented us to make `ChatWindow <=> Discuss` transition as simply same `thread.open()` call with different params to it. It's a pain to maintain all these `markAsRead()` in the whole code by just arbitrary split in function. Hence why I suggest to limit to a single point in code for `markAsRead()`.

This doesn't necessarily just has to be trivially coupled to component. It could be in component itself but also with some flags at model level to detect that even though these are different components, conceptually user hasn't quit the thread so there's no need to trigger `markAsRead()` feature.